packages/mocha/src/role.js

```
@@ -7,16 +7,24 @@ const { wrap } = require('./mocha');
 
 function roles(getRole, getFilePathTitle) {
   return describe => {
-    return function roles(tags, callback) {
+    return function roles(...args) {
       // must be instanced and not inline
-      // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec#Finding_successive_matches
-      let regex = /(?:^| )#([^ ]+)/g;
+      let callback = args.pop();
+
+      let roles;
+      if (args.length > 1) { // roles as multiple arguments
+        roles = args;
+      } else if (Array.isArray(args[0])) { // roles as array in first argument
+        roles = args[0];
+      } else if (typeof args[0] === 'string' || args[0] instanceof String) { // roles as concatenated string
+        let regex = /(?:^| )#([^ ]+)/g;
+        let matches = args[0].matchAll(regex);
+        roles = [...matches].map(i => i[1]);
+      }
```

In case there is an invalid syntax like:

```
  roles(123, function() {
    it('works', function() {
      assert.ok(this.role.match(/role[6|7]/));
    });
  });
```

or

```
  roles('foo bar', function() {
    it('works', function() {
      assert.ok(this.role.match(/role[6|7]/));
    });
  });
```

or whatever else: Do we want to throw an error of some sort? Would that be simple throw?